common/deftype/defvalue.cpp

```diff
@@ -2022,12 +2022,18 @@ unsigned RealValue::getHash(unsigned initval)
 
 const char *RealValue::generateECL(StringBuffer &s)
 {
-    return getStringValue(s);
+    if (isinf(val))
+        return s.append("INFINITY");
```

Comment:
Not sure what generateECL is used for - I was thinking we might introduce a new keyword INFINITY (and maybe one for NAN too)...